other/palindrome.py

```diff
@@ -1,5 +1,25 @@
 # Program to find whether given string is palindrome or not
 def is_palindrome(str):
```

Comment:
`str` is a builtin in Python so this usage "shadows" that builtin which is not a good idea.  Can you please change all occurrences of `str` with just plain `s`?

Also, please change the function signature by adding Python type hints: `def is_palindrome(s: str) -> bool:` and add similar type hints to the function `recursive_palindrome()`.

Let's add another function:
```python
def slice_palindrome(s: str) -> bool:
    return s == s[::-1]
```